### PR TITLE
Fix build previewctl step

### PR DESCRIPTION
**Pull request by geropl**
## Description

Based on https://github.com/gitpod-io/gitpod/pull/18228/.

## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -96,16 +95,24 @@ jobs:
       previewctl_hash: ${{ steps.build.outputs.previewctl_hash }}
     steps:
       - uses: actions/checkout@v3
+      - name: Configure workspace
+        run: |
+          # Needed by docker/login-action
+          sudo chmod goa+rw /var/run/docker.sock
```
Comment (aledbf) [inline .github/workflows/build.yml:101]: What error do you see without this?

Comment (geropl) [reply .github/workflows/build.yml:101]: https://github.com/gitpod-io/gitpod/actions/runs/5519828597/jobs/10065651199#step:4:58

But this is just an attempt, no green build yet.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -96,16 +95,24 @@ jobs:
...
         id: build
         shell: bash
         env:
-          PREVIEW_ENV_DEV_SA_KEY: ${{ secrets.GCP_CREDENTIALS }}
-          LEEWAY_SEGMENT_KEY: "${{ secrets.LEEWAY_SEGMENT_KEY }}"
           LEEWAY_WORKSPACE_ROOT: ${{ env.GITHUB_WORKSPACE }}
+          LEEWAY_SEGMENT_KEY: ""
+          PREVIEW_ENV_DEV_SA_KEY: ${{ secrets.GCP_CREDENTIALS }}
         run: |
+          export PREVIEW_ENV_DEV_SA_KEY_PATH="$GITHUB_WORKSPACE/.config/gcloud/preview-environment-dev-sa.json"
```
Comment (aledbf) [inline .github/workflows/build.yml:110]: Please move `LEEWAY_WORKSPACE_ROOT: ${{ env.GITHUB_WORKSPACE }}` to run
`export LEEWAY_WORKSPACE_ROOT=${{ env.GITHUB_WORKSPACE }}`

**Pull request merged**